src/main/scala/strawman/collection/immutable/Array.scala

```diff
@@ -0,0 +1,84 @@
+package strawman.collection.immutable
+
+import strawman.collection.mutable.{ArrayBuffer, ArrayBufferView, Iterator}
+import strawman.collection.{Iterable, IterableFactory, IterableOnce, IndexedSeq, Seq, SeqLike}
+
+import scala.{AnyRef, Boolean, Int}
+import scala.Predef.{???, intWrapper}
+
+/**
+  * An immutable array.
+  *
+  * Supports efficient indexed access and has a small memory footprint.
+  */
+class Array[+A] private (private val elements: scala.Array[AnyRef]) extends IndexedSeq[A] with SeqLike[A, Array] {
```

Comment:
This needs a public constructor or companion `apply` to make it really useful. Wrapping an existing array without copying is important for performance. Wrapping with copying should also be supported (and be the safe default, of course).

To enable wrapping and copying, you need to take primitive arrays into account as well (like the array-like types in my specialization PR do).